### DISCUSSION TITLE
How to get GPS time as telemetry data with MSP query?

### DISCUSSION THREAD

**Discussion by ucanAlp** [iNavFlight/inav#7401] [Q&A] [answered]
****

## Current Behavior
I can get a lot of telemetry data like roll,pitch,yaw,altitude,longtitude, other gps status variables etc. I'm sending MSP queries from arduino to omnibus F4 V6 with INAV. But I can't get GPS time. When I try to take last massage date with "MSP_GPSSTATISTICS" query it's always returning 200. And it is corresponding Jan 06,1980. I'm using GPS SE100 (M8N GPS).

> 
case MSP_GPSSTATISTICS:
        sbufWriteU16(dst, gpsStats.lastMessageDt);
        sbufWriteU32(dst, gpsStats.errors);
        sbufWriteU32(dst, gpsStats.timeouts);
        sbufWriteU32(dst, gpsStats.packetCount);
        sbufWriteU16(dst, gpsSol.hdop);
        sbufWriteU16(dst, gpsSol.eph);
        sbufWriteU16(dst, gpsSol.epv);
        break;

I'm taking the first parameter(lastMessageDt) and its returning always 200.

## Steps to Reproduce

1.From arduino generating a MSP query with MSP_GPSSTATISTICS(166)
2.Sending it to INAV over serial.
3.Reading receiving reply.
4.It's always 200. Only sometimes 201,202,203

## Expected behavior
Get a reply like a time stamp which is corresponding date with millisecond 

## Suggested solution(s)
I guess there should be another MSP query.

## Additional context

---

- FC Board name and vendor: OMNIBUS F4 V6 
- INAV version string:
# INAV/OMNIBUSF4V6 3.0.1 Jul  4 2021 / 13:38:43 (753c4479)
# GCC-9.3.1 20200408 (release)

**Comment by DzikuVx** [comment, marked as answer]
Try `MSP_RTC` frame

**Reply by mblabla** [reply]
Greattings!!!
I had almost the same question but i was wondering if it could be possible to return from the FC to the TX  the DATE AND TIME as a sensor.
i am using ExpressLRS with the Jumper T16
